### DISCUSSION TITLE
Understanding local fallback for relays

### DISCUSSION THREAD

**Discussion by niallperks** [Blackymas/NSPanel_HA_Blueprint#506] [General]
I seem to be in the small minority of people who still want to use the buttons on the NSPanel to control the relays directly, rather than other functions.

I first enabled the local fallback for each relay button, however this did not activate the relay.

In my automation defined that a button press will toggle the relay, however this relies on the panel being connected to the wifi and home assistant running. I would assume that just enabling the local fallback will process the relay request locally.

Am I missing something or is this the best method to activate the relays using the buttons?

**Comment by deejaybeam** [comment]
This is exactly the use-case for local fallback:
You have to assign the exposed relais to the buttons in the blueprint - this will work as long the panel is connected via the api to HA.
If the panel detects that the HA-Api is offline (HA down, network down, ...) local fallback (if enabled) switches the relais via button-press (locally)

**Comment by niallperks** [comment]
Thanks for the clarification, I was expecting the physical buttons to still toggle the relay without using the blueprint to say 'toggle the relay'

Also tested by taking my HA instance offline and the fallback works as expected

**Comment by Sebfrom17** [comment]
I have another small issue with the local fallback function. In my setup, there is a zigbee lamp connected to the relay of nspanel. To operate it via home assistant the relay must be (and is normally always) on all the time. As it is the main lamp in the room, local fallback is enabled. When wifi (or HA) is off, I can still operate the lamp (at least switch it on/off) via relay. But when I leave it off (relay off) and the wifi recovers, the lamp becomes inoperable (the button is not toggling the relay then, but the zigbee light, which is cut off). Is it possible, that the relay, after wifi (or HA) recovers, comes back to its previous state (before fallback function activated)?
